10up/simple-page-ordering
# Create GitHub JS lint action

### Description of the Change

* Add a GitHub action to run the lining of the JS files
* Add all the required ESLint-related files
* Perform the necessary modifications to the `assets/js/src/simple-page-ordering.js` (the only active JS project file) to pass the linting tests.


Closes #130

### How to test the Change

* While running `npm run dev`, make changes to the `assets/js/src/simple-page-ordering.js` that would generate JS lint errors and verify that the errors are showing in the terminal.
* Make changes to the `assets/js/src/simple-page-ordering.js` that would generate JS lint errors and run `npm run js-lint` and verify that the errors are showing in the terminal.
* Push changes in a PR with some JS lint errors and verify that the JS linting job runs with a failure and produces the proper results (JS lint errors are supposed to be presented in the `Files changed` tab)
* Push changes in a PR without any JS lint errors and verify that the JS linting job runs successfully

### Changelog Entry
* Added - GitHub action to run JS Lint tests.


### Credits

Props @Sidsector9, @kmgalanakis


### Checklist:
